sql/store/niposts.go

```diff
@@ -0,0 +1,126 @@
+package store
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+const (
+	NIPostChallengeKey    = "NIPost"
+	NIPostBuilderStateKey = "NIPostBuilderState"
+)
+
+// AddNIPostChallenge adds the data for nipost to the key-value store.
+func AddNIPostChallenge(db sql.Executor, ch *types.NIPostChallenge) error {
```

Comment:
can you combine niposts.go and commitment.go in on file?
each sql pkg now represents one table and has one file. what's the rationale to deviate from current pattern?

Reply:
I agree with every table one package, but not with every package only has one file.

By splitting up the code into multiple files it is imho easier to read the code and understand what it does, since everything related to each other is in the same file and files stay relatively small.

Imho having 20 files here with 3-5 functions each is much easier to parse as a developer than one file with > 1000 lines of code.

Reply:
sure. but like you said, this is a table that store key value pairs. using your argument, we should have 3 files, each serves a different key for this table? IMO it's over kill. but ok.

Reply:
You're right, I should split up the NiPoST and NiPoSTBuilder.

I tried to make `store`/`get`/`clear` generic but didn't find an approach that I was satisfied with. As an example:

```golang
type key string

const (
   CommitmentATX key = key("CommitmentATX")
   NiPost key = key("NiPost")
)

func StoreValueForKey(db sql.Executor, key key, value codec.Encodable) error

// or

func StoreValueForKey[T codec.Encodable](db sql.Executor, key key, value T) error
```

Both allow for any value to be written for a key to the DB as long as it implements the `codec.Encodable` interface. We could check during runtime if the provided value is of the type we expect via reflection but at that point I thought that this is becoming more complicated than it needs to be and errors would still only be caught during runtime and not compile time.

Reply:
as the code stands now, i don't understand the logical divide of these two files.

i think for me, it is simpler if this table is declared to be a kv table, and that caller simply store []byte inside. it's up to the caller to decide how to encode/decode the data. this table simply stores random kv pair.